{
	"documenttitle": {
		"text": "Red Hat Security Advisory: Red Hat Single Sign-On 7.4.9 security update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2021:3534"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2021-09-14T12:37:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2021-09-14T12:37:00Z",
		"CurrentReleaseDate": "2021-09-14T12:37:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.0.2",
			"Date": "2021-09-14T17:11:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nA security update is now available for Red Hat Single Sign-On 7.4 from the Customer Portal.\n\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nRed Hat Single Sign-On 7.4 is a standalone server, based on the Keycloak project, that provides authentication and standards-based single sign-on capabilities for web and mobile applications.\n\nThis release of Red Hat Single Sign-On 7.4.9 serves as a replacement for Red Hat Single Sign-On 7.4.8, and includes bug fixes and enhancements, which are documented in the Release Notes document linked to in the References.\n\nSecurity Fix(es):\n\n* undertow: buffer leak on incoming websocket PONG message may lead to DoS (CVE-2021-3690)\n\n* jackson-dataformat-cbor:  Unchecked allocation of byte buffer can cause a java.lang.OutOfMemoryError exception (CVE-2020-28491)\n\n* keycloak: Brute force attack is possible even after the account lockout (CVE-2021-3513)\n\n* undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS (CVE-2021-3597)\n\n* keycloak: Anyone can register a new device when there is no device registered for passwordless login (CVE-2021-3632)\n\n* keycloak-model-infinispan: authenticationSessions map in RootAuthenticationSessionEntity grows boundlessly could lead to a DoS attack (CVE-2021-3637)\n\n* jakarta-el: ELParserTokenManager enables invalid EL expressions to be evaluate (CVE-2021-28170)\n\n* apache-commons-io: Limited path traversal in Apache Commons IO 2.2 to 2.6 (CVE-2021-29425)\n\n* keycloak: X509 Direct Grant Auth does not verify certificate timestamp validity (CVE-2020-35509)\n\n* wildfly-core: Invalid Sensitivity Classification of Vault Expression (CVE-2021-3644)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2021 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Moderate",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2021:3534",
				"Description": "https://access.redhat.com/errata/RHSA-2021:3534"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#moderate",
				"Description": "https://access.redhat.com/security/updates/classification/#moderate"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?product=core.service.rhsso&downloadType=securityPatches&version=7.4",
				"Description": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?product=core.service.rhsso&downloadType=securityPatches&version=7.4"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/html/release_notes/index",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.4/html/release_notes/index"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "jackson-dataformat-cbor:  Unchecked allocation of byte buffer can cause a java.lang.OutOfMemoryError exception",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-02-18T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2020-28491",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONDATAFORMAT-1047329",
						"Description": "https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONDATAFORMAT-1047329"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2020-28491",
						"Description": "CVE-2020-28491"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1930423",
						"Description": "bz#1930423: CVE-2020-28491 jackson-dataformat-cbor:  Unchecked allocation of byte buffer can cause a java.lang.OutOfMemoryError exception"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-02-18T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "keycloak: X509 Direct Grant Auth does not verify certificate timestamp validity",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-01-04T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2020-35509",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2020-35509",
						"Description": "CVE-2020-35509"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1912427",
						"Description": "bz#1912427: CVE-2020-35509 keycloak: X509 Direct Grant Auth does not verify certificate timestamp validity"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2021-01-04T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank Luca Leonardo Scorcia for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "keycloak: Brute force attack is possible even after the account lockout",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-04-26T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3513",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3513",
						"Description": "CVE-2021-3513"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1953439",
						"Description": "bz#1953439: CVE-2021-3513 keycloak: Brute force attack is possible even after the account lockout"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-04-26T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank MichaÅ Knapik & Peter Nicholson (u2i) for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-06-11T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3597",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3597",
						"Description": "CVE-2021-3597"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1970930",
						"Description": "bz#1970930: CVE-2021-3597 undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-02-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "5",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "keycloak: Anyone can register a new device when there is no device registered for passwordless login",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-06-12T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3632",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3632",
						"Description": "CVE-2021-3632"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1978196",
						"Description": "bz#1978196: CVE-2021-3632 keycloak: Anyone can register a new device when there is no device registered for passwordless login"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-06-20T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank Florian Ritterhoff (University of Applied Sciences Munich) for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "6",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "keycloak-model-infinispan: authenticationSessions map in RootAuthenticationSessionEntity grows boundlessly could lead to a DoS attack",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2020-12-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3637",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3637",
						"Description": "CVE-2021-3637"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1979638",
						"Description": "bz#1979638: CVE-2021-3637 keycloak-model-infinispan: authenticationSessions map in RootAuthenticationSessionEntity grows boundlessly could lead to a DoS attack"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-07-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank Halil Ä°brahim Åener (hibsener@gmail.com) (Picnic Technologies) for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "7",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "wildfly-core: Invalid Sensitivity Classification of Vault Expression",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-07-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3644",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3644",
						"Description": "CVE-2021-3644"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1976052",
						"Description": "bz#1976052: CVE-2021-3644 wildfly-core: Invalid Sensitivity Classification of Vault Expression"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2021-02-24T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "This issue was discovered by Darran Lofthouse (Red Hat)."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "8",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: buffer leak on incoming websocket PONG message may lead to DoS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-07-30T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3690",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3690",
						"Description": "CVE-2021-3690"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1991299",
						"Description": "bz#1991299: CVE-2021-3690 undertow: buffer leak on incoming websocket PONG message may lead to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-08-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "9",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "jakarta-el: ELParserTokenManager enables invalid EL expressions to be evaluate",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-04-01T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-28170",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://securitylab.github.com/advisories/GHSL-2020-021-jakarta-el/",
						"Description": "https://securitylab.github.com/advisories/GHSL-2020-021-jakarta-el/"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-28170",
						"Description": "CVE-2021-28170"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1965497",
						"Description": "bz#1965497: CVE-2021-28170 jakarta-el: ELParserTokenManager enables invalid EL expressions to be evaluate"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-05-26T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "10",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "apache-commons-io: Limited path traversal in Apache Commons IO 2.2 to 2.6",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-04-12T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-29425",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link (you must log in to download the update).\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2021:3534"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-29425",
						"Description": "CVE-2021-29425"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1948752",
						"Description": "bz#1948752: CVE-2021-29425 apache-commons-io: Limited path traversal in Apache Commons IO 2.2 to 2.6"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-04-12T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {}
}
